test/features/app/spec.js

```diff
@@ -125,8 +125,43 @@ function constructTests(paths, defParams) {
 
 function cmp(result, expected, errMsg) {
 
-    expected = expected || '';
-    result = result || '';
+    if(expected === null || expected === undefined) {
+        // nothing to expect, so we can return
+        return true;
+    }
+    if(result === null || result === undefined) {
+        result = '';
+    }
+
+    if(expected.constructor === Object) {
+        Object.keys(expected).forEach(function(key) {
```

Comment:
It's going to error out a line or two later for sure, which will print out the key name in question. I'm not checking it here explicitly because `cmp()` is recursive, and in `validateTestResponse()` I explicitly convert between the types and check for equivalence there.